server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalScriptedMetricTests.java

```diff
@@ -91,7 +91,7 @@ protected InternalScriptedMetric createTestInstance(String name, Map<String, Obj
     }
 
     private List<Object> randomAggregations() {
-        return randomList(randomBoolean() ? 1 : 5, this::randomAggregation);
+        return randomList(1, randomBoolean() ? 1 : 5, this::randomAggregation);
```

Comment:
I don't think this is the right fix. The tests that fail are the one that always expect a reduced aggregation so they require a single element. That's why `InternalAggregationTestCase#createTestInstanceForXContent` was added but it's not used in `AggregationsTests`. See https://github.com/elastic/elasticsearch/blob/f20699ead53b8912918f4bda6cbfe89a61b8b983/server/src/test/java/org/elasticsearch/search/aggregations/AggregationsTests.java#L299 where replacing the call with `createTestInstanceForXContent` fixes all the failures.

Reply:
Yeah. I *think* the correct fix is in AggregationsTests.java to replace:
```
            aggs.add(testCase.createTestInstance());
```
with:
```
            aggs.add(testCase.createTestInstanceForXContent());
```

I had long forgotten about this test!

Reply:
ups, thanks @jimczi you are right. I guess we will need to make that method accessible to the test as currently is protected.